spec/src/main/doc/events.asciidoc

```diff
@@ -288,18 +318,31 @@ The event parameter type may contain a type variable or wildcard.
 
 The event parameter may be an array type whose component type contains a type variable or a wildcard.
 
+Modifications made to the event parameter in an observer method are propagated to following observers and notification point.
+The container is not required to guarantee a consistent state for an event parameter modified by asynchronous observers.
```

Comment:
Previous line was for synchronous observer, this one is to state that container doesn't provide thread safety for payload modified in async observer.
The idea is to say "if user manage to have a thread safe payload" modification will be taken into account".
Better formulation is of course welcome.